src/util/DarkModeController.js

```diff
@@ -0,0 +1,161 @@
+﻿// @ts-check
```

Comment:
May I recommend you publish this as its own module? :) I'm sure many other projects would enjoy a lightweight dark mode management util.